src/index.coffee

```diff
@@ -0,0 +1,126 @@
+client = require './utils/client'
+
+# utility functions
+checkError = (error, response, body, code, callback) ->
+    callback errorMaker error, response, body, code
+
+errorMaker = (error, response, body, expectedCode) ->
+    if error
+        return error
+    else if response.status isnt expectedCode
+        msgStatus = "expected: #{expectedCode}, got: #{response.statusCode}"
+        err = new Error "#{msgStatus} -- #{body.error} -- #{body.reason}"
+        err.status = response.statusCode
+        return err
+    else
+        return null
+
+define = (docType, name, request, callback) ->
+    {map, reduce} = request
+
+    # transforms all functions in anonymous functions
+    # function named(a, b){...} --> function (a, b){...}
+    # function (a, b){...} --> function (a, b){...}
+    if reduce? and typeof reduce is 'function'
+        reduce = reduce.toString()
+        reduceArgsAndBody = reduce.slice reduce.indexOf '('
+        reduce = "function #{reduceArgsAndBody}"
+
+    view =
+        reduce: reduce
+        map: """
+            function (doc) {
+                if (doc.docType.toLowerCase() === "#{docType.toLowerCase()}") {
+                    filter = #{map.toString()};
+                    filter(doc);
+                }
+            }
+        """
+
+    path = "request/#{docType}/#{name.toLowerCase()}/"
```

Comment:
The data-system doesn't seem to handle well the uppercase/lowercase cases but it will in the future. So i guess that it's only temporary.